server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -811,7 +825,22 @@ public VectorData parseKnnVector(
             XContentParser.Token token = context.parser().currentToken();
             return switch (token) {
                 case START_ARRAY -> parseVectorArray(context, dims, dimChecker, similarity);
-                case VALUE_STRING -> parseHexEncodedVector(context, dimChecker, similarity);
+                case VALUE_STRING -> {
+                    String s = context.parser().text();
+                    if (s.length() == dims * 2 && isMaybeHexString(s)) {
```

Comment:
Is it worth doing the check on each character here? Maybe just try to parse it as hex straight away if its the right length?

Reply:
@thecoop 🤔 hmm, likely that is good enough, especially since I already wrap with a try{}catch(...), this was part of an earlier iteration where I wasn't retrying on failure.